mscgen: fix build

siriobalmelli opened the pull request:
###### Motivation for this change

Found `nix-env -iA "nixpkgs.mscgen"` build was broken because of missing `pkg-config` dependency. Added it.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

siriobalmelli committed:
mscgen: add pkg-config build-time dependency

Signed-off-by: Sirio Balmelli <sirio@b-ad.ch>

calbrecht reviewed (changes requested):
Why are there changes in pkgs howdoi? Seems like this should go into a different PR.

siriobalmelli commented:
> Why are there changes in pkgs howdoi? Seems like this should go into a different PR.

My apologies. They were in fact part of https://github.com/NixOS/nixpkgs/pull/85579 and I screwed up my branching.

Removed these entirely.

calbrecht reviewed (changes requested):
`bison`, `pkg-config` and `flex` should go into `nativeBuildInputs` instead.

calbrecht commented:
Then there is this notice when the tests beeing run
```
Warning: Optional UTF-8 byte-order-mark detected at start of input, but mscgen
         was not configured to use FreeType for text rendering.  Rendering of
         UTF-8 characters in PNG output may be incorrect.
```
Since you added `freetype` you might want to add 
```
configureFlags = [ "--with-freetype" ]
```
and fix the  `Error: gdoTextWidth: Problem doing text layout (GDFONTPATH=)`
or simply leave `freetype` out of dependencies.

calbrecht commented:
And the title of this PR should read like "mscgen: <your description>".

Mic92 commented on pkgs/tools/graphics/mscgen/default.nix:
```diff
@@ -12,7 +24,7 @@ stdenv.mkDerivation {
   };
 
-  buildInputs = [ flex bison gd libjpeg libpng freetype zlib libwebp ];
+  buildInputs = [ bison flex freetype gd libjpeg libpng libwebp pkg-config zlib ];
```
bison, flex and pkg-config should be in `nativeBuildInputs`.

calbrecht replied:
o.O :)

calbrecht commented:
And last, i guess. Consider adding `outputs = [ "out" "man" ];`. This way the man pages are linked correctly.

siriobalmelli committed:
mscgen: clean up build

1. Remove freetype dependency:
    - was not being used by build
    - trying to enable freetype gives nontrivial error
      'gdoTextWidth: Problem doing text layout (GDFONTPATH=)'

2. Correctly link manpages by specifying 'outputs'.

3. Separate nativeBuildInputs from buildInputs

Signed-off-by: Sirio Balmelli <sirio@b-ad.ch>

siriobalmelli commented:
> And last, i guess. Consider adding `outputs = [ "out" "man" ];`. This way the man pages are linked correctly.

Thank you; added in 3a7dca99

siriobalmelli commented:
> `bison`, `pkg-config` and `flex` should go into `nativeBuildInputs` instead.

Thank you, fixed in 3a7dca99

siriobalmelli commented:
> Then there is this notice when the tests beeing run
> 
> ```
> Warning: Optional UTF-8 byte-order-mark detected at start of input, but mscgen
>          was not configured to use FreeType for text rendering.  Rendering of
>          UTF-8 characters in PNG output may be incorrect.
> ```
> 
> Since you added `freetype` you might want to add
> 
> ```
> configureFlags = [ "--with-freetype" ]
> ```
> 
> and fix the `Error: gdoTextWidth: Problem doing text layout (GDFONTPATH=)`
> or simply leave `freetype` out of dependencies.

Thank you.
I gave this a shot but no quick win in sight, so removed freetype dependency entirely in 3a7dca99

calbrecht reviewed (approved):
Good, thank you :+1:

Mic92 commented:
Result of `nixpkgs-review pr 85580` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
<br>- asciidoc-full-with-plugins
<br>- mscgen
</details>
